hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AEndpointRegion.java

```diff
@@ -257,6 +275,33 @@ public void testWithVPCE() throws Throwable {
     expectInterceptorException(client);
   }
 
+  @Test
+  public void testCentralEndpointCrossRegionAccess() throws Throwable {
+    describe("Create bucket on different region and access it using central endpoint");
+    Configuration conf = getConfiguration();
+    removeBaseAndBucketOverrides(conf, ENDPOINT, AWS_REGION);
+
+    Configuration newConf = new Configuration(conf);
+
+    newConf.set(ENDPOINT, CENTRAL_ENDPOINT);
+
+    newFS = new S3AFileSystem();
+    newFS.initialize(getFileSystem().getUri(), newConf);
+
+    final String file = getMethodName();
```

Comment:
I agree with your suggestion but on the other hand, do you think it's okay to keep the test as is so that regardless of the endpoint/region combinations used by anyone while running the suite, at least this test will cover overriding fs.s3a.endpoint with central and performing full file system CRUD operations? My purpose is only to ensure that i can cover as much basic fs ops as possible with just one test so that devs don't have to specifically set central endpoint and run the whole suite (it's a bonus anyways). As of now, the test also covers 400 vs 404 error issue that we came across while running whole suite.

However, that's just my view, happy to make changes as per your comment if you disagree, no issues :)

Reply:
ack, that makes sense, thanks for explaining. happy to keep as is